client/src/components/AdvancedSearch.js

```diff
@@ -237,7 +246,9 @@ const AdvancedSearch = ({
 
   function changeObjectType(objectType) {
     setObjectType(objectType)
-    setFilters([])
+    setFilters(
+      filters.filter(value => commonFiltersForAllObjectTypes[value.key])
+    )
```

Comment:
Have you considered the solution where we keep filter types which are common between previous object type and next object type? For example:
- I have "People" object toggled, i selected "Location", "Organization", "Status" filters. (Which are also in the "Position" object)
- I click "Position" toggle button. "Location" and "Organization" is gone, but "Status" is still there.
 We could keep all the common ones between switching object types.